pkg/scheduler/framework/plugins/noderesources/resource_allocation.go

```diff
@@ -51,7 +51,7 @@ func (r *resourceAllocationScorer) score(
 	if node == nil {
 		return 0, framework.NewStatus(framework.Error, "node not found")
 	}
-	if r.resourceToWeightMap == nil {
+	if len(r.resourceToWeightMap) == 0 {
```

Comment:
Just a personal opinion：I think this is a way to protect the robustness of the code. Although it’s initialized before in all the places to call this function. But it's not mandatory.

Reply:
We have defaults for nil resources, so actually this will not happen I think.